keps/sig-auth/4601-authorize-with-selectors/README.md

```diff
@@ -0,0 +1,795 @@
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The authorization attributes will be extended to include field selectors and label selectors from
+Get, List, Watch, and Deletecollection.
+This will allow authorizers to use these selectors when making an authorization decision.
+
+## Motivation
+
+Security for per-node workloads could be improved by exposing field and label selectors to authorizers.
+Adding them as authorization attributes allows the development of new kinds of authorizers that
+leverage this information to provide security.
+In particular, it enables out-of-tree authorizers to experiment with ways to express restrictions based on field and label selectors.
+
+### Goals
+
+* Add field and label selectors to authorization attributes for Get, List, Watch, and DeleteCollection verbs.
+* Add field and label selectors to webhook authorization types.
+* Add field and label selectors to SelfSubjectAccessReview (SSAR) and SubjectAccessReview (SAR).
+  <<[UNRESOLVED deads2k, liggitt, aws-Micah: you guys volunteering to deliver this bit? ]>>
+* Update node authorizer to restrict on nodeName field selector.
+  <<[/UNRESOLVED]>>
+
+### Non-Goals
+
+* Create a generic in-tree authorizer that manages field or label selectors.
+
+## Proposal
+
+List, Watch, and DeleteCollection requests directly have field and label selector options.
+A single-item List or Watch request is remapped to the Get verb, so Get requests can have field and label selectors.
```

Comment:
I can see a huge value for being able to authorize based on `metadata.name` field (so field selectors).

What is the usecase for the labelSelector? Is it effectively added only for downstream extensibility, or do you see any upstream usecase too? If it's the former, it would be useful to add it explicitly to the KEP.

Reply:
> Is it effectively added only for downstream extensibility, or do you see any upstream usecase too?

Downstream extensibility of APIs by non-API owners.